src/Permissions/Widgets/SidebarRow.vala

```diff
@@ -29,9 +29,16 @@ public class Permissions.SidebarRow : Gtk.ListBoxRow {
     }
 
     construct {
+        hexpand = true;
         var appinfo = new GLib.DesktopAppInfo (app.id + ".desktop");
 
-        var image = new Gtk.Image.from_gicon (appinfo.get_icon (), Gtk.IconSize.DND);
+        Gtk.Image image;
+        if (appinfo != null && appinfo.get_icon () != null) {
+            image = new Gtk.Image.from_gicon (appinfo.get_icon (), Gtk.IconSize.DND);
+        } else {
+            image = new Gtk.Image.from_gicon (new ThemedIcon ("application-vnd.flatpak.ref"), Gtk.IconSize.DND);
```

Comment:
Since what we're representing to the user here is an app and not a file on the filesystem, we should probably use `application-default-icon`. And also we can do this much more simply since Gtk.Image has a built in icon name constructor :)

```suggestion
            image = new Gtk.Image.from_icon_name ("application-default-icon", Gtk.IconSize.DND);
```

Reply:
OK, done.